src/marshmallow/schema.py

```diff
@@ -512,13 +514,16 @@ def _serialize(self, obj: typing.Any, *, many: bool = False):
             ret[key] = value
         return ret
 
-    def dump(self, obj: typing.Any, *, many: bool | None = None):
+    def dump(
+        self, obj: typing.Any, *, many: bool | None = None, context: typing.Any = None
```

Comment:
question: what's the rationale behind adding `context` to `dump` and `load`? i'm not sure the added API surface is worth the convenience.